### PR TITLE
Update htscodecs module to release 1.1.1

### PR DESCRIPTION
**Pull request by daviesrob**
Fixes a bug with name tokeniser and variable number of names per slice.

Adds NEWS items.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -63,6 +63,8 @@ Build changes
 
 These are compiler, configuration and makefile based changes.
 
+* HTSlib now uses libhtscodecs release 1.1.1
```
Comment (jmarshall) [inline NEWS:66]: Fullstop? (or was that intentional as there's so many dots in the last word already :smile:)

File: NEWS [hunk 2]
```diff
@@ -112,6 +114,14 @@ Bug fixes
...
   (#1298, fixes samtools/samtools#1445)
 
+* Bug fixes in the bundled copy of htscodecs:
+
+  - Fixed an uninitialized access in the name tokenizer decoder.
+    (samtools/htscodecs#23)
+
+  - Fixed a bug with name tokeniser and variable number of names per slice,
+    causing it to incorrectly report an error on certain valid inputs.
+    (samtools/htscodecs#24)
```
Comment (jmarshall) [inline NEWS:125]: Usually two blank lines before the previous release.

File: NEWS (hunk 1 above)
Comment (daviesrob) [reply NEWS]: Kind-of due to the number of dots, but I guess I may as well go for it.

File: NEWS (hunk 2 above)
Comment (daviesrob) [reply NEWS:125]: Adjusted...

Comment (jmarshall) [reply NEWS:125]: Sorry for the picky comments :)  As a penance I have checked out the PR to run a spelling checker on it, and noticed it's got `tokeniser` spelt once each way (s vs z)…

Comment (jkbonfield) [reply NEWS:125]: All part of our drive for greater inclusivity ;-)

Comment (daviesrob) [reply NEWS:125]: Oops, yes.  It should use `tokeniser` I guess, as that's the spelling in the library.

**Pull request merged**